### PR TITLE
Make filter_by_language deterministic

### PR DESCRIPTION
**Pull request by warenlg**
Signed-off-by: Waren Long <waren@sourced.tech>

Fixed https://github.com/src-d/lookout-sdk-ml/issues/62

### CODE REVIEW

File: lookout/core/lib.py [hunk 1]
```diff
@@ -75,18 +75,18 @@ def extract_changed_nodes(root: Node, lines: Sequence[int]) -> List[Node]:
...
     """
     Sorts files by programming language and path.
 
     :param files: Iterable of `File`-s.
-    :return: Dictionary with languages as keys and files mapped to paths as values.
+    :return: Dictionary with languages as keys and sorted files by path as values.
     """
-    result = defaultdict(dict)
-    for file in files:
+    result = defaultdict(list)
```
Comment (zurk) [inline lookout/core/lib.py]: Actually, we can use `OrderedDict` from `collections` and I think it is worth to use it for language level also:
```python
result = OrderedDict()  # type: OrderedDict[str, OrderedDict[str, str]]
```

this way we avoid changing the logic in style-analyzer.

**Comment by vmarkovtsev** [comment]
\+1 to @m09's suggestion though: there are a few places where we iterate by language.

**Comment by warenlg** [comment]
I did not find a better solution to initialize an ordered `defaultdict` of `OrderedDict`, I you think about something else, let me know.

File: lookout/core/lib.py [hunk 2]
```diff
@@ -82,11 +82,14 @@ def files_by_language(files: Iterable[File]) -> Dict[str, Dict[str, File]]:
     :param files: Iterable of `File`-s.
     :return: Dictionary with languages as keys and files mapped to paths as values.
     """
-    result = defaultdict(dict)
-    for file in files:
+    result = OrderedDict()
+    for file in sorted(files, key=lambda x: x.path.lower()):
```
Comment (m09) [inline lookout/core/lib.py]: the `.lower()` here might in some rare cases create non-determinism (`/home/mog/A` and `/home/mog/a` could be returned in different orders). It's probably safer to remove it.

Comment (warenlg) [reply lookout/core/lib.py]: ok thanks !

**Pull request merged**